dpctl/tensor/_ctors.py

```diff
@@ -1244,20 +1273,37 @@ def zeros_like(
     if device is None and sycl_queue is None:
         device = x.device
     sycl_queue = normalize_queue_device(sycl_queue=sycl_queue, device=device)
-    sh = x.shape
     dtype = dpt.dtype(dtype)
-    return zeros(
-        sh,
-        dtype=dtype,
-        order=order,
-        device=device,
-        usm_type=usm_type,
-        sycl_queue=sycl_queue,
-    )
+    x_flags = x.flags
+    f_contig = x_flags["F"]
+    c_contig = x_flags["C"]
+    if order == "A":
+        order = "F" if f_contig and not c_contig else "C"
+    if order == "K" and (f_contig or c_contig):
+        order = "C" if c_contig else "F"
```

Comment:
This is a common block of code. It should factored out into `_normalize_order(order, x)` with definition:

```python
def _normalize_order(order, x):
    x_flags = x.flags
    f_contig = x_flags["F"]
    c_contig = x_flags["C"]
    if order == "A":
        order = "F" if f_contig and not c_contig else "C"
    if order == "K" and (f_contig or c_contig):
        order = "C" if c_contig else "F"
    return order
```

and used throughout

Reply:
Good point!